File: 0009-palindrome-number/0009-palindrome-number.c
bool isPalindrome(int x) {
     long int m=x,rev=0;
    while(m>0){
        int d=m%10;
        rev=rev*10+d;
        m/=10;
    }
    if(rev<-2147483648 || rev>2147483648) rev=0;
    if(rev==x) return 1;
    else return 0;
}

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int originalNumber = x;
        int reverse = 0;
        while(x>0){
            int remainder = x%10;
            reverse = reverse*10+remainder;
            x/=10;
        }
        boolean isPalindrome = false;
        if(originalNumber == reverse) isPalindrome = true;
        return isPalindrome;
    }
}